src/gleam/http/request.gleam

```diff
@@ -173,20 +173,34 @@ pub fn set_method(req: Request(body), method: Method) -> Request(body) {
   Request(..req, method: method)
 }
 
-/// A request with commonly used default values. This request can be used as
-/// an initial value and then update to create the desired request.
+/// Create a request from a provided URI string. 
+pub fn new(url: String) -> Request(String) {
+  let assert Ok(parsed_uri) = uri.parse(url)
+  let assert Ok(req) = from_uri(parsed_uri)
+  req
+}
+
+// Helper method for constructing a Request from a url String.
+fn from_url(url: String) -> Result(Request(String), Nil) {
+  use parsed_uri <- result.then(uri.parse(url))
+  from_uri(parsed_uri)
+}
+
+/// Construct a default get request from a provided url.
 ///
-pub fn new() -> Request(String) {
-  Request(
-    method: Get,
-    headers: [],
-    body: "",
-    scheme: http.Https,
-    host: "localhost",
-    port: option.None,
-    path: "",
-    query: option.None,
-  )
+pub fn get(url: String) -> Result(Request(String), Nil) {
+  use req <- result.map(from_url(url))
+  req
+  |> set_method(Get)
+}
+
+/// Construct a default post request from a provided url and body.
+///
+pub fn post(url: String, body: body) -> Result(Request(body), Nil) {
+  use req <- result.map(from_url(url))
+  req
+  |> set_method(Post)
+  |> set_body(body)
```

Comment:
Remove these please 🙏

Reply:
but I like them :(

Reply:
We don't have multiple trivial ways of doing the same thing in Gleam.